### PR TITLE
Fixes for issues found by Coverity static code analysis

### PR DESCRIPTION
**Pull request by edenhill**
https://scan.coverity.com/projects/edenhill-librdkafka?tab=overview

Let's get this merged after 1.2.0

### CODE REVIEW

File: src/rdbuf.c [hunk 1]
```diff
@@ -1167,7 +1167,8 @@ void rd_buf_dump (const rd_buf_t *rbuf, int do_hexdump) {
                         rbuf->rbuf_segment_cnt);
                 TAILQ_FOREACH(seg, &rbuf->rbuf_segments, seg_link) {
                         rd_segment_dump(seg, "  ", 0, do_hexdump);
-                        rd_assert(++segcnt <= rbuf->rbuf_segment_cnt);
+                        segcnt++;
+                        rd_assert(segcnt <= rbuf->rbuf_segment_cnt);
```
Comment (mhowlett) [inline src/rdbuf.c]: the old way seems marginally better in that it's terser? but either/or.

Comment (edenhill) [reply src/rdbuf.c]: Thing is some asserts may be disabled at compile time, in which case the increment wouldn't happen at all.
So better safe than sorry here.

File: src/rdkafka_broker.c [hunk 2]
```diff
@@ -1676,7 +1676,10 @@ int rd_kafka_socket_cb_generic (int domain, int type, int protocol,
         if (s == -1)
                 return -1;
 #ifdef FD_CLOEXEC
-        fcntl(s, F_SETFD, FD_CLOEXEC, &on);
+        if (fcntl(s, F_SETFD, FD_CLOEXEC, &on) == -1)
+                fprintf(stderr, "WARNING: librdkafka: %s: "
```
Comment (mhowlett) [inline src/rdkafka_broker.c]: can you do something else with logging here? people using a custom librdkafka log handler will not see this.

File: src/rdkafka_cgrp.c [hunk 3]
```diff
@@ -3402,6 +3404,9 @@ void rd_kafka_cgrp_handle_SyncGroup (rd_kafka_cgrp_t *rkcg,
         if (rkbuf)
                 rd_kafka_buf_destroy(rkbuf);
 
+        if (assignment)
+                rd_kafka_topic_partition_list_destroy(assignment);
```
Comment (mhowlett) [inline src/rdkafka_cgrp.c]: nice catch

File: src/rdkafka_idempotence.c [hunk 4]
```diff
@@ -163,7 +163,7 @@ int rd_kafka_idemp_request_pid (rd_kafka_t *rk, rd_kafka_broker_t *rkb,
                 return 1;
         }
 
-        rd_kafka_wrunlock(rkb->rkb_rk);
+        rd_kafka_wrunlock(rk);
```
Comment (mhowlett) [inline src/rdkafka_idempotence.c]: do you also want to make this change in the if block above?

File: src/rdkafka_broker.c (hunk 2 above)
Comment (edenhill) [reply src/rdkafka_broker.c]: Yeah that'd be good but unfortunately we don't have access to the rk here.
This warning is just a best effort thing to let the user know CLOEXEC failed, which should be extremely rare, if possible at all.

File: src/rdkafka_idempotence.c (hunk 4 above)
Comment (edenhill) [reply src/rdkafka_idempotence.c]: good catch!

**Pull request merged**